pkg/ext-proc/backend/provider.go

```diff
@@ -130,3 +130,37 @@ func (p *Provider) refreshPodsOnce() error {
 	p.podMetrics.Range(mergeFn)
 	return nil
 }
+
+func (p *Provider) refreshMetricsOnce() error {
```

Comment:
This is moved from metrics.go, no new code

Reply:
The name of this file is slightly confusing. It reads as if it's intended to be a factory class, but then we have metrics related funcs, while there is a metrics.go file. 

What is this file intended to do?

Reply:
yeah probably the name is too broad. It intends to "provide info of the backend", thus "provider", and metrics is part of the "info".